qiskit/providers/honeywell/credentials/__init__.py

```diff
@@ -42,69 +43,90 @@ class HoneywellCredentialsError(HoneywellError):
     pass
 
 
-def discover_credentials(qiskitrc_filename=DEFAULT_QISKITRC_FILE):
-    """ Automatically discover credentials from qiskitrc file or environment variables """
-    creds = None
-    readers = (
-        (read_creds_from_environ, {}),
-        (read_creds_from_qiskitrc, {'filename': qiskitrc_filename})
-    )
-    for reader_func, kwargs in readers:
+class Credentials:
+    """ Class implementing all credential handling for id/refresh tokens """
+    def __init__(self,
+                 token: str = None,
+                 proxies: dict = None):
+        """
+        Credentials Constructor
+        """
+        # Default empty config
+        self.token = None
+        self.proxies = None
+
+        # Load configuration from env/config file
+        self.load_config(DEFAULT_QISKITRC_FILE)
+
+        # Allow overrides if provided
+        if token is not None:
+            self.token = token
+        if proxies is not None:
+            self.proxies = proxies
+
+    def load_from_environ(self):
+        """ Attempts to read credentials from environment variable """
+        self.token = os.getenv('HON_QIS_API')
+
+    def load_from_qiskitrc(self, filename):
+        """ Attempts to read credentials from qiskitrc file
+        The default qiskitrc location is in ``$HOME/.qiskitrc/qhprc``
```

Comment:
I realize this was like this before, but isn't this `$HOME/.qiskit/qhprc`?

```suggestion
        The default qiskitrc location is in ``$HOME/.qiskit/qhprc``
```

Reply:
Good catch.  I'll fix this shortly.